bluesky/callbacks/best_effort.py

```diff
@@ -111,9 +111,21 @@ def start(self, doc):
             dimensions = GUESS  # Fall back on our GUESS.
             warn("We are ignoring the dimensions hinted because we cannot "
                  "combine streams.")
+
         # TODO : Add option to select multiple dim_fields
         self.dim_fields = [fields[0]
                            for fields, stream_name in dimensions]
+
+        # make distinction between flattened fields and plotted fields
+        # motivation for this is that when plotting, we find dependent variable
+        # by finding elements that are not independent variables
+        # TODO : add list comprehension
+        self.independent_vars = list()
```

Comment:
We (@danielballan @stuartcampbell and I) spent a fair amount of time talking about the names on this and settled on 'dimensions' over 'independent' and 'exogenous',  I would rather not re-introduce that ambiguity.   Could this be 

```python
self._no_plot = {f for fields, stream in dimensions for f in fields}
```

?

Reply:
there should be two types of fields: the independent vars we want to plot and the ones we don't want to plot. how about dim_fields and dim_fields_selected?